scene/2d/path_2d.cpp

```diff
@@ -378,9 +378,10 @@ real_t PathFollow2D::get_progress() const {
 }
 
 void PathFollow2D::set_progress_ratio(real_t p_ratio) {
-	if (path && path->get_curve().is_valid() && path->get_curve()->get_baked_length()) {
-		set_progress(p_ratio * path->get_curve()->get_baked_length());
-	}
+	ERR_FAIL_COND_MSG(!path, "Can't set progress ratio on a PathFollow2D that is not part of the scene tree");
```

Comment:
```suggestion
	ERR_FAIL_NULL_MSG(path, "Can't set progress ratio on a PathFollow2D that is not part of the scene tree.");
```
Dedicated check, also formatting

Reply:
thanks for this feedback.

I was not aware of `ERR_FAIL_NULL_MSG`; it indeed makes more sense to use it 👍 

About this:
> it should say that it's not the child of a path instead I'd say

I'm afraid it's a bit more complex than that, because `PathFollow2D->path` is set only when this node enters the scene tree (see https://github.com/godotengine/godot/blob/master/scene/2d/path_2d.cpp#L259 ).
Which means that if I set a `PathFollow2D` as a child of a `Path2D` but then forget to add the `Path2D` in my scene tree, I'll face this behavior were set is silently doing nothing.
But you're correct: `path` may be null if the parent is not a `Path2D`, so I guess the message could be reworded as: 

> Can only set progress ratio on a PathFollow2D that is the child of a Path2D which is itself part of the scene tree.

Does that make more sense like this?

Reply:
That sounds good